bin/run.sh

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env bash
+
+set -e
+
+unset CDPATH
+cd "$( dirname "${BASH_SOURCE[0]}" )/.."
+
+echo() { builtin echo "[SCITRAN] $@"; }
+
+
+set -o allexport
+
+if [ "$#" -eq 1 ]; then
+    EXISTING_ENV=$(env | grep "SCITRAN_" | cat)
+    source "$1"
+    eval "$EXISTING_ENV"
+fi
+if [ "$#" -gt 1 ]; then
+    echo "Usage: $0 [config file]"
+    exit 1
+fi
+
+
+# Default config values
+if [ -z "$SCITRAN_RUNTIME_HOST" ]; then
+    SCITRAN_RUNTIME_HOST="127.0.0.1"
+fi
+if [ -z "$SCITRAN_RUNTIME_PORT" ]; then
+    SCITRAN_RUNTIME_PORT="8080"
+fi
+if [ -z "$SCITRAN_RUNTIME_PATH" ]; then
+    SCITRAN_RUNTIME_PATH="./runtime"
+fi
+if [ -z "$SCITRAN_RUNTIME_SSL_PEM" ]; then
+    SCITRAN_RUNTIME_SSL_PEM=""
+fi
+if [ -z "$SCITRAN_RUNTIME_BOOTSTRAP" ]; then
+    SCITRAN_RUNTIME_BOOTSTRAP="bootstrap.json"
+fi
+if [ -z "$SCITRAN_PERSISTENT_PATH" ]; then
+    SCITRAN_PERSISTENT_PATH="./persistent"
+fi
+if [ -z "$SCITRAN_PERSISTENT_DATA_PATH" ]; then
+    SCITRAN_PERSISTENT_DATA_PATH="$SCITRAN_PERSISTENT_PATH/data"
+fi
+if [ -z "$SCITRAN_PERSISTENT_DB_PATH" ]; then
+    SCITRAN_PERSISTENT_DB_PATH="$SCITRAN_PERSISTENT_PATH/db"
+fi
+if [ -z "$SCITRAN_PERSISTENT_DB_PORT" ]; then
+    SCITRAN_PERSISTENT_DB_PORT="9001"
+fi
+if [ -z "$SCITRAN_PERSISTENT_DB_URI" ]; then
+    SCITRAN_PERSISTENT_DB_URI="mongodb://localhost:$SCITRAN_PERSISTENT_DB_PORT/scitran"
+fi
```

Comment:
Thanks for the one-liners. Much cleaner.

The different default variables serve different purposes. All runtime and most persistent variables only exists in `run.sh`. The two that are duplicated *do not* need to be kept in sync, but are required for the api to operate, if started without external configuration. Hence, their presence in the api. Generally, the api supports being started in multiple ways, with `run.sh` being one of them, without any hard configuration requirements.